File: src/main/java/com/kevin/Hello.java
package com.kevin;

public class Hello {
    public static void main(String[] args) {
//        System.out.println("Hello World!");
        Person p = new Person("tom",66.5f,1.7f);
        p.hello();

        String ab = "abcde";
        System.out.println(ab.charAt(4));
        System.out.println(ab.length());
        System.out.println(ab.substring(1,4));

        int score = 88;
        System.out.println(score < 60 || score > 90);

        // java的char屬於整數型態，因此可以與int做比較，kotlin則無法
        char a = 'A';
        System.out.println(a > 60);
     //   p.weight = 66.5f;
     //   p.height = 1.7f;
        System.out.println(p.bmi());
    }
}

File: src/main/kotlin/com/kevin/Hello.kt
package com.kevin

fun main() {
//    println("Hello Kotlin")
//    Human().hello()
    var a: String? = "abcde"
     //a = null
    println(a?.length)
    println(a?.substring(2,3))
    println(a?.get(3))

    val h = Human(height = 1.7f,weight = 66.5f)
    h.hello()
    println(h.bmi())

    val score = 88
    println(score > 60)

    val c : Char = 'A'
    println(c.toInt() > 60)

//    var age = 19
//    var height = 170.5f
//    var name : String
//    name = "kevin"

}

    class Human(var name : String="", var weight : Float, var height : Float){
    init { //建構子完成式自動執行
        println("your weight is $weight kg ,and your height is $height m.")
    }
//    constructor(name : String, height: Float, weight: Float)
    fun bmi(): Float{
        val bmi = weight / (height*height)
        return bmi
    }
    fun hello(){
        println("Hello Kotlin")
    }
}